client.go

```diff
@@ -136,8 +136,10 @@ func (c *Config) newHTTPClient() (*http.Client, error) {
 		err  error
 	)
 
-	if cert, err = tls.LoadX509KeyPair(c.CertFile, c.KeyFile); err != nil {
-		return nil, errors.Wrapf(err, "failed to read certificate and private key")
+	if c.CertFile != "" && c.KeyFile != "" {
+		if cert, err = tls.LoadX509KeyPair(c.CertFile, c.KeyFile); err != nil {
```

Comment:
this is not enough, as later it will use nil a cert if not set
```
Certificates: []tls.Certificate{cert},
```
let's do sth like

```
certs := []tls.Certificate{}

if (...) {
  certs = append(certs, cert)
}

...

Certificates: certs,
```